flytekit/common/translator.py

```diff
@@ -0,0 +1,260 @@
+            interface=entity.typed_interface,
+            custom={},
+            container=None,
+        )
+        # Reset id to ensure it matches user input
+        cp_entity._id = entity.id
+        cp_entity._has_registered = True
+        cp_entity.assign_name(entity.reference.id.name)
+
+    elif isinstance(entity, ReferenceWorkflow):
+        workflow_metadata = WorkflowMetadata(on_failure=WorkflowFailurePolicy.FAIL_IMMEDIATELY)
+
+        cp_entity = SdkWorkflow(
+            nodes=[],  # Fake an empty list for nodes,
+            id=entity.reference.id,
+            metadata=workflow_metadata,
+            metadata_defaults=workflow_model.WorkflowMetadataDefaults(),
+            interface=entity.typed_interface,
+            output_bindings=[],
+        )
+        # Make sure we don't serialize this
+        cp_entity._has_registered = True
+        cp_entity.assign_name(entity.id.name)
+        cp_entity._id = entity.id
+
+    elif isinstance(entity, ReferenceLaunchPlan):
+        wf_id = _identifier_model.Identifier(_identifier_model.ResourceType.WORKFLOW, "", "", "", "")
+        cp_entity = SdkLaunchPlan(
+            workflow_id=wf_id,
+            entity_metadata=_launch_plan_models.LaunchPlanMetadata(schedule=None, notifications=[]),
+            default_inputs=interface_models.ParameterMap({}),
+            fixed_inputs=literal_models.LiteralMap({}),
+            labels=_common_models.Labels({}),
+            annotations=_common_models.Annotations({}),
+            auth_role=_common_models.AuthRole(assumable_iam_role="fake:role"),
+            raw_output_data_config=RawOutputDataConfig(""),
+        )
+        # Because of how SdkNodes work, it needs one of these interfaces
+        # Hopefully this is more trickery that can be cleaned up in the future
+        cp_entity._interface = TypedInterface.promote_from_model(entity.typed_interface)
+        cp_entity._id = entity.id
+
+        # Make sure we don't serialize this
+        cp_entity._has_registered = True
+        cp_entity.assign_name(entity.reference.id.name)
+
+    elif isinstance(entity, PythonTask):
+        cp_entity = SdkTask(
+            type=entity.task_type,
+            metadata=entity.metadata.to_taskmetadata_model(),
+            interface=entity.interface,
+            custom=entity.get_custom(settings),
+            container=entity.get_container(settings),
+        )
+        # Reset just to make sure it's what we give it
+        cp_entity.id._project = settings.project
+        cp_entity.id._domain = settings.domain
+        cp_entity.id._name = entity.name
+        cp_entity.id._version = settings.version
+
+        if fast:
+            # Containerless tasks are always fast registerable without modification so only operate on tasks that
+            # have a container. Also, raw ContainerTask's should never be touched.
+            if cp_entity.container is not None and not isinstance(entity, ContainerTask):
```

Comment:
hm it makes me sad we lose the inheritence bit. i wonder if it's worth refactoring `get_container` to accept a bool for whether the context should be regular or fast

Reply:
interesting, also, maybe we should do this for any entity that is instance of PythonAutoContainerTask and not ContainerTask. We should assume all container tasks are independent of code?

Reply:
Can we do this in a later PR?  As indicated by Miguel's recent question, I want to see if there are additional use cases for changing/overriding the command of a container.  Even if not, I'd rather clean up in a separate PR since it's not user facing.